libraries/node-core-library/src/Terminal/ITerminalProvider.ts

```diff
@@ -2,13 +2,29 @@
 // See LICENSE in the project root for license information.
 
 /**
+ * Similar to many popular logging packages, terminal providers support a range of message
+ * severities. These severities have built-in formatting defaults in the Terminal object
+ * (warnings are yellow, errors are red, etc.).
+ *
+ * Terminal providers may choose to suppress certain messages based on their severity,
+ * or to route some messages to other providers or not based on severity.
+ *
+ *   Severity  | Purpose
+ *   --------- | -------
+ *   error     | Build errors and fatal issues in rush
+ *   warning   | Not necessarily fatal, but indicate a problem the user should fix
+ *   log       | Informational messages from the rush system
+ *   verbose   | Additional information that may not always be necessary
+ *   debug     | Highest detail level, best used for troubleshooting information
```

Comment:
```suggestion
 *   error     | Build errors and fatal issues
 *   warning   | Not necessarily fatal, but indicate a problem the user should fix
 *   log       | Informational messages
 *   verbose   | Additional information that may not always be necessary
 *   debug     | Highest detail level, best used for troubleshooting information
```

`node-core-library` isn't specifically tied to Rush.